scipy/stats/_multivariate.py

```diff
@@ -806,18 +810,52 @@ def fit(self, x):
             Maximum likelihood estimate of the covariance matrix
 
         """
-        # input validation
+        # input validation for data to be fitted
         x = np.asarray(x)
         if x.ndim != 2:
             raise ValueError("`x` must be two-dimensional.")
 
         n_vectors, dim = x.shape
 
+        # convenience variables for checking if parameters are fixed
+        mean_fixed = fix_mean is not None
+        cov_fixed = fix_cov is not None
+
+        # if both parameters are fixed, there is nothing to fit
+        if mean_fixed and cov_fixed:
+            error_msg = "Both parameters are fixed. There is nothing to fit."
+            raise ValueError(error_msg)
+
         # parameter estimation
         # reference: https://home.ttic.edu/~shubhendu/Slides/Estimation.pdf
-        mean = x.mean(axis=0)
-        centered_data = x - mean
-        cov = centered_data.T @ centered_data / n_vectors
+        if mean_fixed:
+            # input validation for `fix_mean`
+            fix_mean = np.asarray(fix_mean)
+            if fix_mean.ndim != 1:
+                raise ValueError("`fix_mean` must be one-dimensional.")
+            if fix_mean.shape[0] != dim:
+                msg = ("`fix_mean` must be of the same length as "
+                       "the vectors `x`.")
+                raise ValueError(msg)
+            mean = fix_mean
+        else:
+            mean = x.mean(axis=0)
+        if cov_fixed:
+            # input validation for `fix_cov`
+            fix_cov = np.asarray(fix_cov)
+            if fix_cov.ndim != 2:
+                raise ValueError("`fix_cov` must be two-dimensional.")
+            cov_rows, cov_cols = fix_cov.shape
+            if cov_rows != cov_cols:
+                raise ValueError("`fix_cov` must be a square matrix.")
+            if cov_rows != dim:
```

Comment:
Same here. One shape check instead of three separate checks.

However, this also needs a check for positive semidefiniteness.

Reply:
Done.